openhcl/openhcl_boot/src/main.rs

```diff
@@ -484,8 +489,18 @@ mod x86_boot {
             }
         }
 
+        // If TDX-isolated the page tables region of OpenHcl as E820-reserved
+        // as otherwise the L1-VMM can use the pages while APs are spinning
+        // in the reset vector
+        #[cfg(target_arch = "x86_64")]
+        if IsolationType::Tdx == isolation_type {
+            add_e820_entry(entries.next(), page_tables.unwrap(), E820_RESERVED)?;
```

Comment:
does the linux E820 map provided need to be sorted? here it's not probably?

Reply:
I think you actually want to add the pagetables to the reserved slice above. this means maybe reporting it via PartitionInfo or somewhere else. Otherwise, there are other paths where we may think it's RAM (even though it's not)